src/athena.js

```diff
@@ -157,15 +162,26 @@ export let executeQuery = async function({
     throw Error("Athena: query didn't succeed.");
   }
 
-  let queryResult = await athena.getQueryResults({QueryExecutionId}).promise();
+  let queryResult;
+  let nextToken;
+  let rows = [];
+  do {
+    queryResult = await athena.getQueryResults({QueryExecutionId, NextToken: nextToken}).promise();
 
-  if (queryResultIsShowResult(queryResult)) {
-    return queryResultToText(queryResult);
-  }
+    // checking if the query is a result of SHOW query
+    // returing just text in this case, not trying to parse columns and rows
+    if (queryResultIsShowResult(queryResult)) {
+      return queryResultToText(queryResult);
+    }
+
+    nextToken = queryResult.NextToken;
```

Comment:
why not treat pagination as separate case?

* get results
* is this `SHOW`? then return
* are there more rows? then get those as well

Reply:
It could be done like that, but then it will be a bit of code duplication. I don't have much preference with that, so if you insist...